pkg/cmd/version.go

```diff
@@ -0,0 +1,209 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericclioptions/resource"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/must-gather/pkg/util"
+)
+
+var (
+	versionExample = `
+	# Collect debugging data for the "openshift-apiserver-operator"
+	%[1]s version
+`
+)
+
+type VersionOptions struct {
+	printFlags  *genericclioptions.PrintFlags
+	configFlags *genericclioptions.ConfigFlags
+
+	restConfig      *rest.Config
+	kubeClient      kubernetes.Interface
+	discoveryClient discovery.CachedDiscoveryInterface
+	dynamicClient   dynamic.Interface
+
+	podUrlGetter *util.RemotePodURLGetter
+
+	fileWriter *util.MultiSourceFileWriter
+	builder    *resource.Builder
+	args       []string
+
+	// directory where all gathered data will be stored
+	baseDir string
+	// whether or not to allow writes to an existing and populated base directory
+	overwrite bool
+
+	genericclioptions.IOStreams
+}
+
+func NewVersionOptions(streams genericclioptions.IOStreams) *VersionOptions {
+	return &VersionOptions{
+		printFlags:  genericclioptions.NewPrintFlags("gathered").WithDefaultOutput("yaml").WithTypeSetter(scheme.Scheme),
+		configFlags: genericclioptions.NewConfigFlags(),
+		IOStreams:   streams,
+	}
+}
+
+func NewCmdVersion(parentName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewVersionOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:          "version [flags]",
+		Short:        "Gather debugging data for a given Cluser Version",
+		Example:      fmt.Sprintf(versionExample, parentName),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&o.baseDir, "base-dir", "must-gather", "Root directory used for storing all gathered cluster operator data. Defaults to $(PWD)/must-gather")
+	cmd.Flags().BoolVar(&o.overwrite, "overwrite", false, "If true, allow this command to write to an existing location with previous data present")
```

Comment:
You can safely remove this flag. Also, default `o.overwrite` to true under `NewVersionOptions`